fix: Add back partner supplied bios GRO-1170

jonallured opened the pull request:
This PR adds back the partner supplied bios - here's a few takes:

### with a partner supplied bio

<img width="1374" alt="Screen Shot 2022-07-26 at 4 29 52 PM" src="https://user-images.githubusercontent.com/79799/181117029-3cda7de7-cfa5-4b26-aafe-4149939b0474.png">


### with an editorial bio

<img width="1374" alt="Screen Shot 2022-07-26 at 4 30 06 PM" src="https://user-images.githubusercontent.com/79799/181117049-9d9f9208-2d4e-4bb3-8e6c-3f7e99d99d28.png">


### with no bio at all

<img width="1374" alt="Screen Shot 2022-07-26 at 4 30 01 PM" src="https://user-images.githubusercontent.com/79799/181117082-1f306dcd-b769-47a8-bcb4-c6d65a7779e0.png">


https://artsyproduct.atlassian.net/browse/GRO-1170

/cc @artsy/grow-devs

TMcMeans commented:
lgtm but I'm a little confused about where the conditional for rendering partner bios vs generic/no bios is, is that supposed to be `showCredit`?? Also, did we eventually make sense of where the CVs are supposed to go??

jonallured commented:
Good call @TMcMeans I def went too fast and could have added back the tests too - I have done that and this is ready for another review! 🙌 

Your question about conditional rendering is a good one and caused me to make some other changes too. First let's remember that editorial bios run up top next to the artist's name whereas partner bios run under the Notable Works rail. So the conditional logic is split:

* in `ArtistHeader` we check for a partner bio and suppress the editorial bio (here)[https://github.com/artsy/force/blob/jonallured/fix-add-back-partner-supplied-bios/src/Apps/Artist/Components/ArtistHeader/ArtistHeader.tsx#L26]
* then here in this PR i've added a bit to `ArtistCareerHighlights` that checks for the partner bio and runs it if should

That second bit could have used a better name so I switched things around and made a boolean called `showPartnerBio` that checks the same things as before but makes it more clear I think. What do you think?

---

In terms of CV, that's outside the scope of this PR. We did decide that we could add the CV link elsewhere on the page but I'm going to follow up with that in a separate PR.

TMcMeans commented:
I like the `showPartnerBio` boolean being parsed out into a variable it makes things much more readable.
